webapp/src/components/systemConsole/surveyDateTime/surveyDateTime.tsx

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useCallback, useEffect, useState} from 'react';
+import {format, parse} from 'date-fns';
+
+import SurveyTimeSelector from 'components/surveyTimeSelector/surveyTimeSelector';
+import SurveyDateSelector from 'components/surveyDateSelector/surveyDateSelector';
+import {CustomComponentProps} from 'types/mattermost-webapp';
+
+import './style.scss';
+
+const defaultSurveyTime = '09:00';
+
+// date 30 days from today
+const defaultSurveyDate = new Date(new Date().setDate((new Date()).getDate() + 30));
+
+function SurveyDateTime({id, setSaveNeeded, onChange, config}: CustomComponentProps) {
+    const [surveyTime, setSurveyTime] = useState<string>(defaultSurveyTime);
+    const [surveyDate, setSurveyDate] = useState<Date>(defaultSurveyDate);
+
+    useEffect(() => {
+        // sets the date picker and time picker to the values saved in config on load
+
+        const dateTimeConfig = config.PluginSettings?.Plugins?.['com.mattermost.user-survey']?.surveydatetime;
+
+        if (dateTimeConfig?.time) {
+            setSurveyTime(dateTimeConfig.time);
+        }
+
+        if (dateTimeConfig?.date) {
+            setSurveyDate(parse(dateTimeConfig.date, 'dd/MM/yyyy', new Date()));
+        }
+    }, [config.PluginSettings, config.PluginSettings?.Plugins]);
```

Comment:
I think we can improve this dependency as just:
`config.PluginSettings?.Plugins?.['com.mattermost.user-survey']?.surveydatetime`
It is more specific of what we need, so it renders less often. We could argue that this won't change much eitherway in this context and have something this explicit is bad for readability.

In any case, having these two dependencies seems unnecessary. One of them should suffice.

Reply:
Note that doing that might require naming it before it gets to this point. I feel like ESLint would complain that the current code doesn't depend on `config`

Reply:
Having just `config.PluginSettings?.Plugins` was sufficient. And yeah, if I set it to `config.PluginSettings?.Plugins?.['com.mattermost.user-survey']?.surveydatetime` ESlint complains of having too complex expression in dependency array. So I think we can leave it to just `config.PluginSettings?.Plugins`.

Reply:
ESLint will warn, but will not fail. But good practice to store the "actual value you need" in a variable so it is more readable.